src/GridFieldOrderableRows.php

```diff
@@ -507,47 +530,73 @@ protected function executeReorder(GridField $grid, $sortedIDs)
+     * @param array $sortedIDs [newSortValue => listItemID]
+     */
     protected function reorderItems($list, array $values, array $sortedIDs)
     {
+        // setup
         $sortField = $this->getSortField();
-        /** @var SS_List $map */
-        $map = $list->map('ID', $sortField);
-        //fix for versions of SS that return inconsistent types for `map` function
-        if ($map instanceof Map) {
-            $map = $map->toArray();
-        }
+        $class = $list->dataClass();
+        // The problem is that $sortedIDs is a list of the _related_ item IDs, which causes trouble
+        // with ManyManyThrough, where we need the ID of the _join_ item in order to set the value.
+        $itemToSortReference = ($list instanceof ManyManyThroughList) ? 'getJoin' : 'Me';
+        $currentSortList = $list->map('ID', $itemToSortReference)->toArray();
 
-        // If not a ManyManyList and using versioning, detect it.
+        // sanity check.
         $this->validateSortField($list);
-        $isVersioned = false;
-        $class = $list->dataClass();
 
+        $isVersioned = false;
         // check if sort column is present on the model provided by dataClass() and if it's versioned
         // cases:
         // Model has sort column and is versioned - handle as versioned
         // Model has sort column and is NOT versioned - handle as NOT versioned
         // Model doesn't have sort column because sort column is on ManyManyList - handle as NOT versioned
-
-        // try to match table name, note that we have to cover the case where the table which has the sort column
-        // belongs to ancestor of the object which is populating the list
-        $classes = ClassInfo::ancestry($class, true);
-        foreach ($classes as $currentClass) {
-            if (DataObject::getSchema()->tableName($currentClass) == $this->getSortTable($list)) {
-                $isVersioned = $class::has_extension(Versioned::class);
-                break;
+        // Model doesn't have sort column because sort column is on ManyManyThroughList...
+        //   - Related item is not versioned:
+        //       - Through object is versioned: THROW an error.
+        //       - Through object is NOT versioned: handle as NOT versioned
+        //   - Related item is versioned...
+        //       - Through object is versioned: handle as versioned
+        //       - Through object is NOT versioned: THROW an error.
+        $isManyMany = $this->isManyMany($list);
+        if ($isManyMany && $list instanceof ManyManyThroughList) {
+            $listClassVersioned = $class::create()->hasExtension(Versioned::class);
+            // We'll be updating the join class, not the relation class.
+            $class = $this->getManyManyInspector($list)->getJoinClass();
+            $isVersioned = $class::create()->hasExtension(Versioned::class);
+
+            if ($listClassVersioned xor $isVersioned) {
```

Comment:
Is this implying that the base class and a MMT that is owns can't both be versioned?

Reply:
No, the opposite.

Reply:
"if the list class is versioned or (never and) the data class versioned"

I'm confused

Reply:
If one is versioned and the other is not, throw an error.
I'll add a comment :)